x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/CCRIT.java

```diff
@@ -0,0 +1,292 @@
+
+    private static void deleteAutoFollowPattern(String patternName) throws IOException {
+        Request request = new Request("DELETE", "/_ccr/auto_follow/" + patternName);
+        assertOK(client().performRequest(request));
+    }
+
+    private static void index(String index, String id) throws IOException {
+        Request request = new Request("POST", "/" + index + "/_doc/" + id);
+        request.setJsonEntity("{}");
+        assertOK(client().performRequest(request));
+    }
+
+    private static void assertDocumentExists(String index, String id) throws IOException {
+        Request request = new Request("HEAD", "/" + index + "/_doc/" + id);
+        Response response = client().performRequest(request);
+        assertThat(response.getStatusLine().getStatusCode(), equalTo(200));
+    }
+
+    private static void setupRemoteCluster() throws IOException {
+        Request request = new Request("GET", "/_nodes");
+        Map<?, ?> nodesResponse = (Map<?, ?>) toMap(client().performRequest(request)).get("nodes");
+        // Select node info of first node (we don't know the node id):
+        nodesResponse = (Map<?, ?>) nodesResponse.get(nodesResponse.keySet().iterator().next());
+        String transportAddress = (String) nodesResponse.get("transport_address");
+
+        LOGGER.info("Configuring local remote cluster [{}]", transportAddress);
+        request = new Request("PUT", "/_cluster/settings");
+        request.setJsonEntity("{\"persistent\": {\"cluster.remote.local.seeds\": \"" + transportAddress + "\"}}");
+        assertThat(client().performRequest(request).getStatusLine().getStatusCode(), equalTo(200));
+    }
+
+    private int getNumberOfSuccessfulFollowedIndices() throws IOException {
+        Request statsRequest = new Request("GET", "/_ccr/stats");
+        Map<?, ?> response = toMap(client().performRequest(statsRequest));
+        Integer actualSuccessfulFollowedIndices = ObjectPath.eval("auto_follow_stats.number_of_successful_follow_indices", response);
+        if (actualSuccessfulFollowedIndices != null) {
+            return actualSuccessfulFollowedIndices;
+        } else {
+            return -1;
+        }
+    }
+
+    private void assertNumberOfSuccessfulFollowedIndices(int expectedNumberOfSuccessfulFollowedIndices) throws IOException {
+        Request statsRequest = new Request("GET", "/_ccr/stats");
+        Map<?, ?> response = toMap(client().performRequest(statsRequest));
+        LOGGER.info("AUTO FOLLOW STATS={}", response.get("auto_follow_stats"));
+        Integer actualSuccessfulFollowedIndices = ObjectPath.eval("auto_follow_stats.number_of_successful_follow_indices", response);
+        assertThat(actualSuccessfulFollowedIndices, equalTo(expectedNumberOfSuccessfulFollowedIndices));
+    }
+
+    private void assertFollowerGlobalCheckpoint(String followerIndex, int expectedFollowerCheckpoint) throws IOException {
+        Request statsRequest = new Request("GET", "/" + followerIndex + "/_ccr/stats");
```

Comment:
I think a seq_no stats of the following index would give us a stronger confidence than the ccr stats.